### ISSUE TITLE
azurerm_subnet_route_table_association fails with ID contained more segments than required

### ISSUE THREAD

**Issue by camilo-s** [hashicorp/terraform-provider-azurerm#18416]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.9

### AzureRM Provider Version

3.23.0

### Affected Resource(s)/Data Source(s)

azurerm_subnet_route_table_association

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "~>3.23.0"
    }
  }
}


provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "this" {
  name     = "rg-network"
  location = var.location
}

resource "azurerm_virtual_network" "this" {
  name                = "vnet-name"
  location            = azurerm_resource_group.this.location
  resource_group_name = azurerm_resource_group.this.name
  address_space       = var.virtual_network_address_space
}

resource "azurerm_virtual_network_dns_servers" "this" {
  virtual_network_id = azurerm_virtual_network.this.id
  dns_servers        = var.dns_servers
}

resource "azurerm_subnet" "main_subnet" {
  name                 = "snet-name"
  resource_group_name  = azurerm_resource_group.this.name
  virtual_network_name = azurerm_virtual_network.this.name
  address_prefixes     = var.main_subnet_address_prefixes
}

resource "azurerm_route_table" "this" {
  name                          = "rt-name"
  location                      = azurerm_resource_group.this.location
  resource_group_name           = azurerm_resource_group.this.name
  disable_bgp_route_propagation = true
}

resource "azurerm_route" "all_to_firewall" {
  name                   = "all-to-hub"
  resource_group_name    = azurerm_resource_group.this.name
  route_table_name       = azurerm_route_table.this.name
  address_prefix         = "0.0.0.0/0"
  next_hop_type          = "VirtualAppliance"
  next_hop_in_ip_address = var.azure_firewall_ip_address
}

resource "azurerm_subnet_route_table_association" "main_subnet_to_firewall" {
  subnet_id      = azurerm_subnet.main_subnet.id
  route_table_id = azurerm_route.all_to_firewall.id
}
```

### Debug Output/Panic Output

```shell
https://gist.github.com/camilo-s/27ce1add85286b2ffabea6e21b9a122c
```

### Expected Behaviour

All resources should deploy with no problem.

### Actual Behaviour

All resources get properly deployed EXCEPT the `azurerm_subnet_route_table_association` one, which shows the following error:

```
Terraform command 'apply' failed with exit code '1'.
Error: ID contained more segments than required: "/subscriptions/###/resourceGroups/rg-network/providers/Microsoft.Network/routeTables/rt-name/routes/all-to-hub", map[routes:all-to-hub]
```

### Steps to Reproduce

1. `terraform init`
2. `terraform plan`
3. `terraform apply`

### Important Factoids

Curiously, `azurerm_subnet_route_table_association` can be deployed when sourcing the `route_table_id` argument from a data resource as in `route_table_id = data.azurerm_route_table.all_to_firewall.id`

**Comment by sinbai** [comment]
@camilo-s thanks for opening this issue here. Per the [tf doc](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/subnet_route_table_association), the `route_table_id ` of `azurerm_subnet_route_table_association` should be a route table id instead of the id of `azurerm_route`. Could you try updating the `azurerm_subnet_route_table_association` block like below and try again?

```
resource "azurerm_subnet_route_table_association" "main_subnet_to_firewall" {
  subnet_id      = azurerm_subnet.main_subnet.id
  route_table_id = azurerm_route_table.this.id
}
```

**Comment by camilo-s** [comment]
@sinbai Thanks for the pointer and for the quick reply! 
Wou're right, I was assigning the incorrect value to the `route_table_id ` parameter. I'm now able to deploy the subnet route table association.

I guess that, on top of my own slip, I was misled by Terraform's only failing at the `apply` stage and then reporting `Error: ID contained more segments than required`.
Would it be possible to include an input validation of sorts for Terraform to detect such an incorrect value assignment for `route_table_id ` prior to `terraform apply`?

**Comment by sinbai** [comment]
@camilo-s good suggestion. [PR ](https://github.com/hashicorp/terraform-provider-azurerm/pull/18440)has been submitted to fix it.

**Comment by sinbai** [comment]
@camilo-s Since the issue has been fixed in [v3.24.0 of the Terraform Provider](https://github.com/hashicorp/terraform-provider-azurerm/blob/v3.24.0/CHANGELOG.md). Are you okey if we close this issue?

**Comment by tombuildsstuff** [comment]
Closing this out per @sinbai's message above, but if this is still an issue for you in the latest version of the Provider please let us know and we'll take another look.

Thanks!
